packages/guides-restructured-text/src/RestructuredText/Directives/ConfigurationBlockDirective.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\RestructuredText\Directives;
+
+use phpDocumentor\Guides\Nodes\CodeNode;
+use phpDocumentor\Guides\Nodes\CollectionNode;
+use phpDocumentor\Guides\Nodes\Configuration\ConfigurationBlockNode;
+use phpDocumentor\Guides\Nodes\Configuration\ConfigurationTab;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\RestructuredText\Parser\BlockContext;
+use phpDocumentor\Guides\RestructuredText\Parser\Directive;
+use phpDocumentor\Guides\RestructuredText\Parser\Productions\Rule;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\String\Slugger\AsciiSlugger;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+use function assert;
+use function get_debug_type;
+use function str_replace;
+use function ucfirst;
+
+final class ConfigurationBlockDirective extends SubDirective
+{
+    /** @var array<string, string> */
+    private array $languageLabels = [];
+
+    private SluggerInterface $slugger;
+
+    /**
+     * @param Rule<CollectionNode> $startingRule
+     * @param array<array{language: string, label: string}> $languageLabels
+     */
+    public function __construct(
+        private LoggerInterface $logger,
+        Rule $startingRule,
+        array $languageLabels = [],
+    ) {
+        parent::__construct($startingRule);
+
+        $this->slugger = new AsciiSlugger();
+
+        foreach ($languageLabels as $item) {
+            $this->languageLabels[$item['language']] = $item['label'];
+        }
```

Comment:
I would move this to the `RestructuredTextExtension::load()` method and make the argument `array<string, string>` (i.e. "array<language, label>`).